### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by ydirson**
A view generic  improvements and fixes I did while working on zfs-vol tests, but largely unrelated

### CODE REVIEW

**Review by stormi — APPROVED** [review]
All good to me. Some of the added strings might deserve to be make f-strings, but that's not a blocker for me.

**Comment by ydirson** [comment]
> Some of the added strings might deserve to be make f-strings

After a quick review, rather the contrary, one logging call is wrongly using a f-string. The convention is to use lazy formatting on logging, especially so disabled logging levels (debug, trace) will not apply potentially-expensive formatting into strings that are immediately thrown away.

**Review by benjamreis — APPROVED** [review]

**Comment by stormi** [comment]
Lazy formatting hasn't been used in this project for now, so while this is a good argument in absolute, it also introduces more inconsistencies to the project. But I can't claim it was already consistent so again, not a blocker for me.

**Comment by ydirson** [comment]
> one logging call is wrongly using a f-string

pushed an update to make at least the PR consistent on this

**Review by gthvn1 — APPROVED** [review]

File: lib/vdi.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
         # TODO: use a different approach when migration is possible
         self.sr = sr
 
     def destroy(self):
+        logging.info("Destroy %s", self)
         self.sr.pool.master.xe('vdi-destroy', {'uuid': self.uuid})
+
+    def __str__(self):
+        return (f"VDI {self.uuid} on SR {self.sr.uuid}"
+                f"{f' (snapshot of {self.snapshot_of})' if self.snapshot_of else ''}")
```
Comment (stormi) [inline lib/vdi.py:15]: I was too confident with this PR: this made the tests fail

```
tests/storage/ext/test\_ext\_sr.py::TestEXTSR::test\_snapshot\[172.16.210.22-[http://pxe.lyon.vates.team/images/alpine-minimal-3.12.0.xva-auto](http://pxe.lyon.vates.team/images/alpine-minimal-3.12.0.xva-auto)\] ERROR

==================================== ERRORS ====================================
\_ ERROR at teardown of TestEXTSR.test\_snapshot\[172.16.210.22-[http://pxe.lyon.vates.team/images/alpine-minimal-3.12.0.xva-auto](http://pxe.lyon.vates.team/images/alpine-minimal-3.12.0.xva-auto)\] \_

ext\_sr = <lib.sr.SR object at 0x7f2477c4d080>

    @pytest.fixture(scope='module')
    def vdi\_on\_ext\_sr(ext\_sr):
        vdi = ext\_sr.create\_vdi('EXT-local-VDI-test')
        yield vdi
>       vdi.destroy()
```

```
tests/storage/ext/conftest.py:16:
self = <lib.vdi.VDI object at 0x7f2477c4d198>

    def __str__(self):
>       return (f"VDI {self.uuid} on SR {self.sr.uuid}"
                f"{f' (snapshot of {self.snapshot_of})' if self.snapshot_of else ''}")
E       AttributeError: 'VDI' object has no attribute 'snapshot_of'

lib/vdi.py:14: AttributeError
```

**Pull request merged**